docs/versioned_docs/version-0.2.x/how_to/llm_token_usage_tracking.ipynb

```diff
@@ -2,169 +2,228 @@
      "text": [
-      "72\n"
+      "\n",
+      "\n",
+      "Why did the chicken go to the seance?\n",
+      "\n",
+      "To talk to the other side of the road!\n",
+      "--\n",
+      "\n",
+      "\n",
+      "Why did the fish blush?\n",
+      "\n",
+      "Because it saw the ocean's bottom!\n",
+      "\n",
+      "---\n",
+      "Total Tokens: 49\n",
+      "Prompt Tokens: 12\n",
+      "Completion Tokens: 37\n",
+      "Total Cost (USD): $9.200000000000001e-05\n"
      ]
     }
    ],
    "source": [
+    "from langchain_community.callbacks import get_openai_callback\n",
+    "from langchain_core.prompts import PromptTemplate\n",
+    "from langchain_openai import OpenAI\n",
+    "\n",
+    "llm = OpenAI(model_name=\"gpt-3.5-turbo-instruct\")\n",
+    "\n",
+    "template = PromptTemplate.from_template(\"Tell me a joke about {topic}\")\n",
+    "chain = template | llm\n",
+    "\n",
     "with get_openai_callback() as cb:\n",
-    "    result = llm.invoke(\"Tell me a joke\")\n",
-    "    result2 = llm.invoke(\"Tell me a joke\")\n",
-    "    print(cb.total_tokens)"
+    "    response = chain.invoke({\"topic\": \"birds\"})\n",
+    "    print(response)\n",
+    "    response = chain.invoke({\"topic\": \"fish\"})\n",
+    "    print(\"--\")\n",
+    "    print(response)\n",
+    "\n",
+    "\n",
+    "print()\n",
+    "print(\"---\")\n",
+    "print(f\"Total Tokens: {cb.total_tokens}\")\n",
+    "print(f\"Prompt Tokens: {cb.prompt_tokens}\")\n",
+    "print(f\"Completion Tokens: {cb.completion_tokens}\")\n",
+    "print(f\"Total Cost (USD): ${cb.total_cost}\")"
    ]
   },
   {
    "cell_type": "markdown",
-   "id": "d8186e7b",
-   "metadata": {},
-   "source": [
-    "If a chain or agent with multiple steps in it is used, it will track all those steps."
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": 5,
-   "id": "5d1125c6",
-   "metadata": {},
-   "outputs": [],
+   "id": "ad7a3fba-9fac-4222-8f87-d1d276d27d6e",
+   "metadata": {
+    "tags": []
+   },
    "source": [
-    "from langchain.agents import AgentType, initialize_agent, load_tools\n",
-    "from langchain_openai import OpenAI\n",
+    "## Streaming\n",
+    "\n",
+    ":::{.callout-danger}\n",
+    "\n",
+    "`get_openai_callback` relies on metadata information in the API response to get\n",
+    "information about the token counts; however, the `OpenAI` API does **NOT** return\n",
+    "token counts when streaming.\n",
     "\n",
-    "llm = OpenAI(temperature=0)\n",
-    "tools = load_tools([\"serpapi\", \"llm-math\"], llm=llm)\n",
-    "agent = initialize_agent(\n",
-    "    tools, llm, agent=AgentType.ZERO_SHOT_REACT_DESCRIPTION, verbose=True\n",
-    ")"
+    "If you want to count token counts correctly, for streaming, either\n",
+    "implement a custom callback handler that uses an appropriate tokenizers to\n",
+    "count the tokens or use a monitoring platform like [LangSmith](https://www.langchain.com/langsmith).\n",
```

Comment:
Consider adding something like "Note that below no tokens are counted by the callback handler, because this metadata is not currently supplied by OpenAI in streaming mode."